deeptime/markov/msm/tram/_tram_dataset.py

```diff
@@ -0,0 +1,494 @@
+                # place submodel counts in our full-sized count matrices
+                transition_counts[k, i_s, j_s] = model_k.count_matrix.T
+
+        return transition_counts
+
+    @property
+    def state_counts(self):
+        r""" ndarray(n, m)
+        The state counts histogram. state_counts[k] contains the state histogram for thermodynamic state k,
+        based on the TransitionCountModel of state k. state_counts[k][i] equals the number of samples that fall into
+        Markov state :math:`i`, sampled at thermodynamic state :math:`k`.
+
+        The state counts for every thermodynamic state are shaped to contain all possible markov states, even the ones
+        without counts in that thermodynamic state. This is done so that the _tram_bindings receives count matrices that
+        are all the same shape, which is easier to handle (matrices are padded with zeros for all empty states that got
+        dropped by the TransitionCountModels).
+        """
+        state_counts = np.zeros((self.n_therm_states, self.n_markov_states), dtype=np.intc)
+
+        for k in range(self.n_therm_states):
+            model_k = self.count_models[k]
+            if model_k.count_matrix.sum() > 0:
+                state_counts[k, model_k.state_symbols] = model_k.state_histogram
+
+        return state_counts
+
+    def check_against_model(self, model):
+        r""" Check the number of thermodynamic states of the model against that of the dataset. The number of
+        thermodynamic states in the dataset have to be smaller than or equal to those of the model, otherwise the ttrajs
+        and/or dtrajs would be out of bounds.
+
+        Parameters
+        ----------
+        model : TRAMModel
+            The TRAMModel to check the data against.
+
+        Raises
+        ------
+        ValueError
+            If the data encompasses more Markov and/or thermodynamic states than the model.
+        """
+        if model.n_therm_states < self._n_therm_states:
+            if self.ttrajs is None:
+                msg = "dtrajs are out of bounds of the model. " \
+                      "There are more dtrajs then there are thermodynamic states. Provide ttrajs " \
+                      "or increase the number of thermodynamic states."
+            else:
+                msg = "ttrajs are out of bounds of the model. " \
+                      "Largest thermodynamic state index in ttrajs would be larger than the number of thermodynamic " \
+                      "states in the model."
+            raise ValueError(msg)
+        if model.n_markov_states < self._n_markov_states:
+            raise ValueError(
+                "dtrajs are out of bounds of the model."
+                "Largest Markov state index in dtrajs would be larger than the number of Markov states in the model.")
+
+    def restrict_to_largest_connected_set(self, connectivity='post_hoc_RE', connectivity_factor=1.0, progress_bar=None):
```

Comment:
perhaps call `progress` instead of `progress_bar` for consistency with the rest of the lib